src/BTVNanoCommissioning/workflows/ttsemilep_valid_sf.py

```diff
@@ -345,48 +732,77 @@ def process_shift(self, events, shift_name):
+                        h.fill(
+                            syst,
+                            flatten(smu[histname.replace("mu_", "")]),
+                            weight=weight,
+                        )
+                    elif (
+                        "jet" in histname
+                        and "dr" not in histname
+                        and "njet" != histname
+                    ):
+                        if "c" in histname:
+                            print(
+                                "jet in histname",
+                                histname,
+                                scjets[histname.replace(f"cjet_", "")],
+                            )
+                            jet_array = flatten(scjets[histname.replace(f"cjet_", "")])
+                            print(
+                                "Lengths:",
+                                len(syst),
+                                len(flatten(genflavor_c)),
+                                len(jet_array),
+                                len(weight),
+                            )
+
                             h.fill(
                                 syst,
-                                flatten(genflavor[:, i]),
-                                flatten(sel_jet[histname.replace(f"jet{i}_", "")]),
+                                flatten(genflavor_c),
+                                flatten(scjets[histname.replace(f"cjet_", "")]),
                                 weight=weight,
                             )
 
-            for i in range(4):
-                output[f"dr_mujet{i}"].fill(
+                output[f"dr_cjet"].fill(
                     syst,
-                    flav=flatten(genflavor[:, i]),
-                    dr=flatten(smu.delta_r(sjets[:, i])),
+                    flav=flatten(genflavor_c),
+                    dr=flatten(smu.delta_r(scjets)),
                     weight=weight,
                 )
-            output["njet"].fill(syst, nseljet, weight=weight)
-            output["MET_pt"].fill(syst, flatten(smet.pt), weight=weight)
-            output["MET_phi"].fill(syst, flatten(smet.phi), weight=weight)
-            output["npvs"].fill(
-                syst,
-                events[event_level].PV.npvs,
-                weight=weight,
-            )
-            if not isRealData:
-                output["pu"].fill(
+                output["njet"].fill(syst, ncseljet, weight=weight)
+                output["MET_pt"].fill(syst, flatten(smet.pt), weight=weight)
+                output["MET_phi"].fill(syst, flatten(smet.phi), weight=weight)
+                output["npvs"].fill(
+                    syst,
+                    events[event_level].PV.npvs,
+                    weight=weight,
+                )
+                output["w_mass"].fill(
                     syst,
-                    events[event_level].Pileup.nTrueInt,
+                    flatten(sw.mass),
                     weight=weight,
                 )
```

Comment:
the W mass should stored in both MC and data right?

Reply:
Exactly, there is no condition